src/java/org/apache/cassandra/service/CassandraDaemon.java

```diff
@@ -401,6 +402,11 @@ public void startNativeTransport()
             throw new IllegalStateException("thrift transport should be set up before it can be started");
         thriftServer.start();
         logger.info("Thrift server running on {}", new InetSocketAddress(DatabaseDescriptor.getRpcAddress(), DatabaseDescriptor.getRpcPort()));
+
+        if (!CrossVpcIpMappingHandshaker.instance.isEnabled())
```

Comment:
Why is this the right place for this call? Why not when we start gossip, for instance?

Reply:
Also we should double check that this is properly started if we disable/re-enable the appropriate protocols, i.e., when we call `StorageService#startTransports`

Reply:
So you suggest adding it here? https://github.com/palantir/cassandra/blob/bdcda1712fd76196f793fe737f9e683826511905/src/java/org/apache/cassandra/service/StorageService.java#L829

Reply:
my hunch is that you might need to add it everywhere that Gossiper#start is called

Reply:
Added explicitly to Gossiper#start, Gossiper#stop, as well as shadow round

Reply:
why stop?

Reply:
Conceptually it made sense to me to stop sending these handshakes when we stop gossip + want the node to be isolated. Are you suggesting keeping it on all the time even when gossip is down? Either way I don't feel strongly.

Reply:
Ah nvm, I thought you were going to start handshaking on "stop".